Sources/MapboxCoreNavigation/PassiveLocationDataSource.swift

```diff
@@ -65,66 +56,8 @@ open class PassiveLocationDataSource: NSObject {
     /**
      Starts the generation of location updates with an optional completion handler that gets called when the location data source is ready to receive snapped location updates.
      */
-    public func startUpdatingLocation(completionHandler: ((Error?) -> Void)? = nil) {
+    public func startUpdatingLocation() {
         systemLocationManager.startUpdatingLocation()
-        
-        guard !isConfigured else {
-            return
-        }
-        
-        directions.fetchAvailableOfflineVersions { [weak self] (versions, error) in
-            guard let self = self, let latestVersion = versions?.first(where: { !$0.isEmpty }), error == nil else {
-                completionHandler?(error)
-                return
-            }
-            
-            do {
-                try self.configureNavigator(withTilesVersion: latestVersion)
-                completionHandler?(nil)
-            } catch {
-                completionHandler?(error)
-            }
-        }
-    }
-    
-    /**
-     Creates a cache for tiles of the given version and configures the navigator to use this cache.
-     */
-    func configureNavigator(withTilesVersion tilesVersion: String) throws {
```

Comment:
The original idea was to defer the navigator’s creation and configuration until the map SDK starts consuming location updates from PassiveLocationManager. Now that configuration happens earlier when attaching the PassiveLocationManager to the MapView. The nice thing about waiting until `startUpdatingLocation()` is that we could make it completely asynchronous without having to synchronize within a factory method.